L1Trigger/DemonstratorTools/src/codecs_etsums.cc

```diff
@@ -5,11 +5,11 @@ namespace l1t::demo::codecs {
 
   ap_uint<64> encodeEtSum(const l1t::EtSum& etSum) {
     l1tmetemu::EtMiss etMiss;
-    etMiss.Et = etSum.hwPt();
+    ap_uint<l1tmetemu::kMETSize> ET = etSum.hwPt();
```

Comment:
Another, perhaps cleaner way of doing this is:
```c++
etMiss.Et.V = etSum.hwPt();
```
and then changing L12 to be:
```c++
ap_uint<64> etSumWord = (unassigned, etMiss.Phi, etMiss.Et.range(), valid);
```
This way you don't have to create another temporary variable or another arbitrary type.

Reply:
fixed